zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/Schema.scala

```diff
@@ -184,19 +184,31 @@ trait Measurement {
 
 //TODO when UI get the schema, it should know which is dimension/measure, functions that can apply onto it, etc.
 // so that it won't ask for a inapplicable function such that get the max to a string field
-case class Schema(typeName: String,
-                  dimension: Seq[Field],
-                  measurement: Seq[Field],
-                  primaryKey: Seq[Field],
-                  timeField: TimeField
-                 ) {
+abstract class Schema(typeName: String,
+                      dimension: Seq[Field],
+                      measurement: Seq[Field],
+                      primaryKey: Seq[Field]
+                     ) {
```

Comment:
I think it's better to think a little bit how to unify the common parts of TemporalSchema and StaticSchema. Try to reduce the distinguish of them in your codebase, but rather unify them as the abstract schema.
For example, is it OK to add `timeField` as a method of Schema, but only TemporalSchema implements it? Calling `timeField` on StaticSchema would automatically throw an Exception.